### PR TITLE
DOCS-1651: Node Local DNS Cache changes in CC docs

### PR DESCRIPTION
**Pull request by girishshankaran**
Product Version(s):

Issue:
https://tigera.atlassian.net/browse/DOCS-1651

Link to docs preview:

SME review:
- [x] An SME has approved this change. 

DOCS review:
- [x] A member of the docs team has approved this change.

Additional information:

Merge checklist:

- [x] Deploy preview inspected wherever changes were made 
- [x] Build completed successfully
- [x] Test have passed

### CODE REVIEW

**Review by bmckercher123 — APPROVED** [review]
LGTM

File: calico-cloud/networking/configuring/node-local-dns-cache.mdx [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+## Before you begin
+
+### Required
+
+Follow these [steps](https://kubernetes.io/docs/tasks/administer-cluster/nodelocaldns/) to enable NodeLocal DNSCache connectivity.
+
+### Unsupported
+
+- OpenShift
```
Comment (tmjd) [inline calico-cloud/networking/configuring/node-local-dns-cache.mdx]: Should we call out OpenShift as unsupported for this one feature when OpenShift is not supported on Calico Cloud?

Comment (bmckercher123) [reply calico-cloud/networking/configuring/node-local-dns-cache.mdx]: Good catch. This is the CE version of the doc. Yes, we should remove this for CC @girishshankaran. @tmjd But, I don't see that we call out non-support for OCP here: https://docs.tigera.io/calico-cloud/get-started/connect/requirements/system-requirements. Do I need to open a PR to add it?

Comment (Glen-Tigera) [reply calico-cloud/networking/configuring/node-local-dns-cache.mdx]: In addition to non-support for OCP in CC, I think we should also mention we don't support RKE and RKE2 in CC as well

File: calico-cloud/networking/configuring/node-local-dns-cache.mdx [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+Set up NodeLocal DNSCache to improve DNS lookup latency.
+
+## Before you begin
+
+### Required
+
+Follow these [steps](https://kubernetes.io/docs/tasks/administer-cluster/nodelocaldns/) to enable NodeLocal DNSCache connectivity.
+
+### Unsupported
```
Comment (Glen-Tigera) [inline calico-cloud/networking/configuring/node-local-dns-cache.mdx]: I would mention GKE has a known issue running NodeLocalDNS Cache on GKE while upgrading to Calico Cloud if NodeLocalDNS was enabled _before_ upgrading. It works if you enable this feature if you upgrade to cloud first. https://tigera.atlassian.net/browse/EV-3878.

File: calico-cloud/networking/configuring/node-local-dns-cache.mdx (hunk 1 above)
Comment (girishshankaran) [reply calico-cloud/networking/configuring/node-local-dns-cache.mdx]: @bmckercher123 @tmjd @Glen-Tigera I removed the information about OpenShift and just added RKE and RKE2 in the `Unsupported` section.

Comment (bmckercher123) [reply calico-cloud/networking/configuring/node-local-dns-cache.mdx]: @girishshankaran We should remove all references to Unsupported (including RKE and RKE2) in this doc. The upshot of what Erik and Glen are saying is that CC does not support OCP, RKE, and RKE2 as platforms. Period. That needs to be called out in a separate PR to update CC system requirements. I will put up a PR for that.

File: calico-cloud/networking/configuring/node-local-dns-cache.mdx (hunk 2 above)
Comment (bmckercher123) [reply calico-cloud/networking/configuring/node-local-dns-cache.mdx]: @Glen-Tigera Good point. I will add it to the CC RN and let PM decide to publish it or keep it as internal corner case.

File: calico-cloud/networking/configuring/node-local-dns-cache.mdx (hunk 1 above)
Comment (girishshankaran) [reply calico-cloud/networking/configuring/node-local-dns-cache.mdx]: Done

**Review by bmckercher123 — APPROVED** [review]
LGTM

**Pull request merged**